src/lib/statsig.ts

```diff
@@ -0,0 +1,50 @@
+import { StatSigFlags, StatsigConfigType } from '@/types/statsig';
+import { StatsigClient } from '@statsig/js-client';
+
+const fetchIp = async () => {
+  const resp = await fetch('https://api.ipify.org?format=json');
+  return (await resp.json())?.ip;
+};
+
+let statsigClient: StatsigClient;
+
+const initStatsig = async () => {
+  if (statsigClient) return statsigClient;
+  statsigClient = new StatsigClient(
+    import.meta.env.VITE_STATSIG_CLIENT_KEY,
+    {
+      ip: await fetchIp(),
+    },
+    {
+      disableLogging: import.meta.env.VITE_DISABLE_STATSIG,
+      disableStorage: import.meta.env.VITE_DISABLE_STATSIG,
+    }
+  );
+  if (import.meta.env.VITE_DISABLE_STATSIG) return statsigClient;
```

Comment:
hmm should we be returning anything if `DISABLE` is set to true?

might be easier to read as:

```
if (import.meta.env.VITE_DISABLE_STATSIG) {
  return null
} else if (!statsigClient) {
  statsigClient = ...
  await statsigClient.initializeSync()
}

return statsigClient;
```

Reply:
ooo we actually dont even need this early return since we're disabling its comms to the statsig api regardless